delta-api/src/main/java/io/cdap/delta/api/assessment/ColumnDetail.java

```diff
@@ -26,11 +28,20 @@
   private final String name;
   private final SQLType type;
   private final boolean nullable;
+  private final Map<String, String> properties;
 
   public ColumnDetail(String name, SQLType type, boolean nullable) {
     this.name = name;
     this.type = type;
     this.nullable = nullable;
+    this.properties = new HashMap<>();
+  }
+
+  public ColumnDetail(String name, SQLType type, boolean nullable, Map<String, String> properties) {
+    this.name = name;
+    this.type = type;
+    this.nullable = nullable;
+    this.properties = properties;
```

Comment:
To protect against bad plugin behavior, should make a copy of the input map and wrap it in an unmodifiable one.